Cargo.toml

```diff
@@ -34,10 +34,11 @@ default = ["mimalloc"]
 tokio = { version = "1.24", features = ["macros", "rt", "rt-multi-thread", "sync"] }
 rand = "0.8"
 pyo3 = { version = "~0.17.3", features = ["extension-module", "abi3", "abi3-py37"] }
-datafusion = { version = "16.0.0", features = ["pyarrow", "avro"] }
-datafusion-expr = { version = "16.0.0" }
-datafusion-optimizer = { version = "16.0.0" }
-datafusion-common = { version = "16.0.0", features = ["pyarrow"] }
+datafusion = { git = "https://github.com/apache/arrow-datafusion", branch = "master", features = ["pyarrow", "avro"] }
+datafusion-expr = { git = "https://github.com/apache/arrow-datafusion", branch = "master" }
+datafusion-optimizer = { git = "https://github.com/apache/arrow-datafusion", branch = "master" }
+datafusion-common = { git = "https://github.com/apache/arrow-datafusion", branch = "master", features = ["pyarrow"] }
+datafusion-substrait = { git = "https://github.com/apache/arrow-datafusion", branch = "master" }
```

Comment:
Let's pin to a specific git rev here rather than master to keep the master branch here stable. We can PR to increase the rev as often as we need to.

Reply:
Good point. Will change that now